grid2op/tests/test_RewardAlertCostScore.py

```diff
@@ -0,0 +1,201 @@
+        assert simr == 0., f"{simr} vs 0."
+        assert simd
+        
+    def test_simulated_env(self):
+        obs = self.env.reset()
+        f_env = obs.get_forecast_env()
+        forD = False
+        while not forD:
+            forO, forR, forD, forI = f_env.step(self.env.action_space())
+            assert forR == 0.
+            
+        f_env = obs.get_forecast_env()
+        forD = False
+        go_act = self.env.action_space({"set_bus": {"generators_id": [(0, -1)]}})
+        while not forD:
+            forO, forR, forD, forI = f_env.step(go_act)
+            assert forR == 0.
+    
+    
+class TestRunner(unittest.TestCase):
+    def setUp(self) -> None:
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+        self.env = grid2op.make(self.env_nm, test=True, difficulty="1",
+                        reward_class=_AlertCostScore)
+        self.env.seed(0)
+        return super().setUp()
+    
+    def tearDown(self) -> None:
+        self.env.close()
+        return super().tearDown()
+    
+    def test_dn_agent(self):
+        obs = self.env.reset()
+        runner = Runner(**self.env.get_params_for_runner())
+        res = runner.run(nb_episode=1, episode_id=[0], max_iter=10, env_seeds=[0])
+        assert res[0][2] == 1. #it got to the end
+    
+    def test_simagent(self):
+        #simulate blackout but act donothing
+        obs = self.env.reset()
+        
+        class SimAgent(BaseAgent):
+            def act(self, observation: BaseObservation, reward: float, done: bool = False) -> BaseAction:
+                go_act = self.action_space({"set_bus": {"generators_id": [(0, -1)]}})
+                simO, simr, simd, simi = obs.simulate(go_act)
+                simO, simr, simd, simi = obs.simulate(self.action_space())
+                return super().act(observation, reward, done)
+            
+        runner = Runner(**self.env.get_params_for_runner(),
+                        agentClass=SimAgent)
+        res = runner.run(nb_episode=1, episode_id=[0], max_iter=10, env_seeds=[0])
+        assert res[0][2] == 1.
+        
+    def test_episodeData(self):
+        obs = self.env.reset()
+        runner = Runner(**self.env.get_params_for_runner())
+        res = runner.run(nb_episode=1, episode_id=[0], max_iter=10, env_seeds=[0], add_detailed_output=True)
+        assert res[0][2] == 1.
+        assert res[0][5].rewards[8] == 1.
+        
+    def test_with_save(self):
+        obs = self.env.reset()
+        runner = Runner(**self.env.get_params_for_runner())
+        with tempfile.TemporaryDirectory() as f:
+            res = runner.run(nb_episode=1, episode_id=[0], max_iter=10, env_seeds=[0],
+                             path_save=f)
+            assert res[0][2] == 1.
+            ep0, *_ = EpisodeData.list_episode(f)
+            ep = EpisodeData.from_disk(*ep0)
+            assert ep.rewards[8] == 1.
+ 
+
+class TestAlertTrustScore(unittest.TestCase):  
```

Comment:
More tests are needed here I think. First to test the -100 / +100 range is properly met
Then to do some basic tests when there are some attacks and some alerts or not to make sure the computed stuff is properly calibrated.